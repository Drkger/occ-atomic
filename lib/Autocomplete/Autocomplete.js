import React from 'react';
import PropTypes from 'prop-types';

import Input from '../Input';
import Droplist from '../Droplist';
import Label from '../Label';

/**
* This component integrates the Input, Droplist and Label components.
* The value of Input filters the list in Droplist.
* The selected items of Droplist can be stacked into Labels.
*/
class Autocomplete extends React.Component {

    constructor(props) {
        super(props);
        this.state = {
            value: props.inputProps.valueProp ? props.inputProps.valueProp : '',
            focus: false,
            labels: []
        };
        this.onChange = this.onChange.bind(this);
        this.onFocus = this.onFocus.bind(this);
        this.onBlur = this.onBlur.bind(this);
        this.onKeyUp = this.onKeyUp.bind(this);
        this.onClear = this.onClear.bind(this);
        this.onClick = this.onClick.bind(this);
        this.addLabel = this.addLabel.bind(this);
        this.removeLabel = this.removeLabel.bind(this);
        this.removeSelected = this.removeSelected.bind(this);
        this.forceValue = this.forceValue.bind(this);
        this.setRef = this.setRef.bind(this);
        this.labels = [];
    }

    componentWillMount() {
        let { onRef } = this.props;
        if (onRef) onRef(this);
    }

    componentWillUnmount() {
        let { onRef } = this.props;
        if (onRef) onRef(undefined);
    }

    onChange(value) {
        const { onChange } = this.props;
        this.setState({
            value, focus: true
        });
        if (onChange) onChange(value);
    }

    onFocus() {
        const { onFocus } = this.props;
        this.setState({
            focus: true
        });
        if (onFocus) onFocus();
    }

    onBlur() {
        const { onBlur } = this.props;
        setTimeout(() => {
            this.setState({
                focus: false
            });
        }, 200);
        if (onBlur) onBlur();
    }

    onKeyUp(key) {
        const { onKeyUp, selectOnEnter, droplistProps } = this.props;
        if (selectOnEnter) {
            if (key == 13) {
                const cleanItems = this.removeSelected();
                if (cleanItems.length) {
                    if (!droplistProps.groups) {
                        const item = cleanItems[0];
                        this.onClick(item);
                    } else {
                        const item = cleanItems[0][droplistProps.groupItemsKey][0];
                        this.onClick(item);
                    }
                } else {
                    this.setState({
                        value: ''
                    });
                }
            }
        }
        if (onKeyUp) onKeyUp(key);
    }

    onClear() {
        const { onClear } = this.props;
        this.setState({
            value: ''
        });
        if (onClear) onClear();
    }

    onClick(item) {
        const { onClick, onChange, droplistProps, stackLabels } = this.props;
        if (stackLabels) {
            this.input.onClear();
            this.setState({
                value: ''
            });
            this.input.input.focus();
            this.addLabel(item);
        } else {
            this.setState({
                value: item[droplistProps.itemTextKey], focus: false
            });
            if (onChange) onChange(item[droplistProps.itemTextKey]);
        }
        if (onClick) onClick(stackLabels ? this.labels : item);
    }

    addLabel(item) {
        this.labels.push(item);
        this.setState({labels: this.labels});
    }

    removeLabel(item) {
        const { droplistProps, onClick } = this.props;
        let pos = this.labels.map(label => { return label[droplistProps.itemIdKey];}).indexOf(item[droplistProps.itemIdKey]);
        this.labels.splice(pos, 1);
        this.setState({labels: this.labels});
        if (onClick) onClick(this.labels);
    }

    removeSelected() {
        const { labels } = this.state;
        const { droplistProps } = this.props;
        let cleanItems = [];
        if (droplistProps.groups) {
            droplistProps.items.map(group => {
                let cleanGroup = group[droplistProps.groupItemsKey].filter(item => {
                    const inLabels = labels.filter(label => {
                        return label[droplistProps.itemIdKey] == item[droplistProps.itemIdKey];
                    });
                    return !inLabels.length;
                });
                const modifiedGroup = Object.assign({}, group);
                modifiedGroup[droplistProps.groupItemsKey] = cleanGroup;
                if (cleanGroup.length) cleanItems.push(modifiedGroup);
            });
        } else {
            cleanItems = droplistProps.items.filter(item => {
                const inLabels = labels.filter(label => {
                    return label[droplistProps.itemIdKey] == item[droplistProps.itemIdKey];
                });
                return !inLabels.length;
            });
        }
        return cleanItems;
    }

    forceValue(value) {
        this.setState({value});
    }

    setRef(input) {
        if (input)
            this.input = input;
    }

    render() {
        const { value, focus, labels } = this.state;
        const { classes, id, inputProps, droplistProps, stackLabels, limit, limitPlaceholder, className, style, disableAutoComplete } = this.props;
        const cleanItems = this.removeSelected();
        return (
            <div className={`${classes.autoComplete}${className ? ` ${className}` : ''}`} id={id} style={style} ref = {ref => { this.autocomplete = ref; }}>
                <Input
                    onRef={this.setRef}
                    id={inputProps.id}
                    type={inputProps.type}
                    name={inputProps.name}
                    label={inputProps.label}
                    placeholder={((stackLabels && limit) && this.labels.length >= limit) ? limitPlaceholder : inputProps.placeholder}
                    onChange={this.onChange}
                    onFocus={this.onFocus}
                    onBlur={this.onBlur}
                    onKeyUp={this.onKeyUp}
                    onClear={this.onClear}
                    selectOnFocus={inputProps.selectOnFocus}
                    clear={inputProps.clear}
                    disabled={(stackLabels && limit) && this.labels.length >= limit}
                    valueProp={value}
                    iconName={inputProps.iconName}
                    theme={inputProps.theme}
                    className={inputProps.className}
                    inputClassName={inputProps.inputClassName}
                    disableAutoComplete={disableAutoComplete}
                />
                {(value && focus && (cleanItems.length)) ? (
                    <Droplist
                        items={cleanItems}
                        groups={droplistProps.groups}
                        groupNameKey={droplistProps.groupNameKey}
                        groupIdKey={droplistProps.groupIdKey}
                        groupItemsKey={droplistProps.groupItemsKey}
                        itemTextKey={droplistProps.itemTextKey}
                        itemTextRightKey={droplistProps.itemTextRightKey}
                        itemIdKey={droplistProps.itemIdKey}
                        term={value}
                        filter={droplistProps.filter}
                        onClick={this.onClick}
                        className={`${classes.droplist}${droplistProps.className ? ` ${droplistProps.className}` : ''}`} />
                ) : null}
                {(stackLabels && labels.length) ? (
                    <div className={classes.labels}>
                        {
                            labels.map(label => (
                                <Label key={label[droplistProps.itemIdKey]} onClose={() => {this.removeLabel(label);}}>{label[droplistProps.itemTextKey]}</Label>
                            ))
                        }
                    </div>
                ) : ''}
            </div>
        );
    }
}

Autocomplete.propTypes = {
    classes: PropTypes.object,
    /** Object with the props for the Input component. Check their descriptions in the Input docs. */
    inputProps: PropTypes.shape({
        id: PropTypes.string,
        type: PropTypes.string,
        name: PropTypes.string,
        label: PropTypes.string,
        placeholder: PropTypes.string,
        selectOnFocus: PropTypes.bool,
        clear: PropTypes.bool,
        iconName: PropTypes.string,
        theme: PropTypes.string,
        valueProp: PropTypes.string,
        className: PropTypes.string,
        inputClassName: PropTypes.string
    }).isRequired,
    /** Object with the props for the Droplist component. Check their descriptions in the Droplist docs. */
    droplistProps: PropTypes.shape({
        items: PropTypes.array,
        itemIdKey: PropTypes.string,
        itemTextKey: PropTypes.string,
        itemTextRightKey: PropTypes.string,
        groups: PropTypes.bool,
        groupIdKey: PropTypes.string,
        groupNameKey: PropTypes.string,
        groupItemsKey: PropTypes.string,
        className: PropTypes.string,
        filter: PropTypes.bool
    }).isRequired,
    /** Function to call on change (Input). */
    onChange: PropTypes.func,
    /** Function to call on key up (Input). */
    onKeyUp: PropTypes.func,
    /** Function to call on focus (Input). */
    onFocus: PropTypes.func,
    /** Function to call on blur (Input). */
    onBlur: PropTypes.func,
    /** Function to call on clear (Input). */
    onClear: PropTypes.func,
    /** Function to call on click (Droplist item). */
    onClick: PropTypes.func,
    onRef: PropTypes.func,
    /** Set this property to stack the selected items of Droplist into Labels. */
    stackLabels: PropTypes.bool,
    /** Set the limit of stackable items into labels. */
    limit: PropTypes.number,
    /** Change the placeholder when the limit of labels has been reached. */
    limitPlaceholder: PropTypes.string,
    /** Automatically selects the first item of the Droplist when pressing the Enter key. NOTE: To use this property correctly you must set the droplistProps.filter property as false and handle the filtering yourself, since Autocomplete only has access to the droplistProps.items property and not the items already filtered by Droplist. */
    selectOnEnter: PropTypes.bool,
    /**  Disables native autoComplete on browsers. */
    disableAutoComplete: PropTypes.bool,
    id: PropTypes.string,
    className: PropTypes.string,
    style: PropTypes.object
};

Autocomplete.defaultProps = {
    inputProps: {},
    droplistProps: {},
    disableAutoComplete: false
};

export default Autocomplete;
