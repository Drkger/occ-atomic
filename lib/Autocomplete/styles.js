import spacing from '../subatomic/spacing';

export default {
    autoComplete: {
        position: 'relative'
    },
    droplist: {
        position: 'absolute',
        zIndex: 2,
        top: `calc(100% - ${spacing.xTiny}px)`,
        left: 0,
        width: '100%'
    }
};
