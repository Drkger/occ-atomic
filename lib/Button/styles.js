import colors from '../subatomic/colors';
import fonts from '../subatomic/fonts';
import icons from '../subatomic/icons';

export default {
    btn: {
        alignItems:'flex-start',
        boxSizing:'border-box',
        display:'inline-block',
        marginBottom:'0',
        textAlign:'center',
        whiteSpace:'nowrap',
        verticalAlign:'middle',
        touchAction:'manipulation',
        cursor:'pointer',
        userSelect:'none',
        padding:'0 30px',
        height:'30px',
        lineHeight:'24px',
        fontFamily:fonts.body,
        fontSize:'14px',
        fontWeight:'600',
        position:'relative',
        transition:'0.3s all',
        maxWidth:'100%',
        borderRadius:'5px',
        textDecoration:'none',
        '&:focus': {
            outline:'none'
        },
        '&:hover': {
            textDecoration:'none'
        }
    },
    cont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        pointerEvents:'none'
    },
    loadCont: {
        display:'flex',
        width:'100%',
        height:'100%',
        alignItems:'center',
        justifyContent:'center',
        position:'absolute',
        top:'0',
        left:'0',
        pointerEvents:'none'
    },
    // Themes
    main: {
        background:colors.pink,
        color:colors.white,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    secondary: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid ${colors.pink}`,
        '&:hover': {
            background:colors.pinkH,
            color:colors.white,
            borderColor:colors.pinkH,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:colors.pinkA,
            color:colors.white,
            borderColor:colors.pinkA,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '&$loading': {
            background:colors.pinkA,
            borderColor:colors.pinkA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    white: {
        background:'transparent',
        color:colors.white,
        border:`1px solid ${colors.white}`,
        '&:hover': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '&:active': {
            background:'rgba(255,255,255,0.2)',
            color:colors.white,
            borderColor:colors.white,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
            }
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:'rgba(255,255,255,0.2)',
            borderColor:colors.white
        },
        '&$disabled': {
            background:`transparent !important`,
            color:`rgba(255,255,255,0.6) !important`,
            border:`1px solid rgba(255,255,255,0.6) !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon(['rgba(255,255,255,0.6)'])) : ''
            }
        }
    },
    flat: {
        background:'transparent',
        color:colors.pink,
        border:`1px solid transparent`,
        '&:hover': {
            background:colors.grey3,
            color:colors.pink,
            borderColor:colors.grey3,
        },
        '&:active': {
            background:colors.grey5,
            color:colors.pink,
            borderColor:colors.grey5,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.pink])) : ''
        },
        '& $loadIcon': {
            background: icons.base(icons.loading.icon([colors.pink]))
        },
        '&$loading': {
            background:colors.white,
            borderColor:colors.white
        },
        '&$disabled': {
            background:'transparent !important',
            color:`${colors.grey1} !important`,
            border:`1px solid transparent !important`,
            '& i': {
                background: props => props.icon ? icons.base(icons[props.icon].icon([colors.grey1])) : ''
            },
            '& $loadIcon': {
                background: icons.base(icons.loading.icon([colors.grey1]))
            },
        }
    },
    blue: {
        background:colors.blue,
        color:colors.white,
        border:`1px solid ${colors.blue}`,
        '&:hover': {
            background:colors.blueH,
            color:colors.white,
            borderColor:colors.blueH,
        },
        '&:active': {
            background:colors.blueA,
            color:colors.white,
            borderColor:colors.blueA,
        },
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&$loading': {
            background:colors.blueA,
            borderColor:colors.blueA
        },
        '&$disabled': {
            background:`${colors.grey1} !important`,
            color:`${colors.white} !important`,
            border:`1px solid ${colors.grey1} !important`
        }
    },
    loading: {
        pointerEvents:'none',
        '& $cont': {
            visibility:'hidden'
        }
    },
    disabled: {
        background:colors.grey1,
        color:colors.white,
        border:`1px solid ${colors.grey1}`,
        cursor:'default',
        pointerEvents:'none',
        '& i': {
            background: props => props.icon ? icons.base(icons[props.icon].icon([colors.white])) : ''
        },
        '&:hover, &:active, &:focus': {
            background:colors.grey1,
            color:colors.white,
            border:`1px solid ${colors.grey1}`
        }
    },
    // Sizes
    md: {
        fontSize:'16px',
        height:'40px',
        lineHeight:'28px',
        '& i': {
            width:'28px',
            height:'28px'
        }
    },
    lg: {
        fontSize:'18px',
        height:'46px',
        lineHeight:'32px',
        '& i': {
            width:'32px',
            height:'32px'
        }
    },
    // Block
    block: {
        display:'block',
        width:'100%'
    },
    // Icon
    icon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all'
    },
    '@keyframes rotate': {
        from: {transform: 'rotate(0deg)'},
        to: {transform: 'rotate(360deg)'}
    },
    loadIcon: {
        width:'24px',
        height:'24px',
        display:'inline-block',
        transition:'0.3s all',
        animation:'rotate 1s infinite linear',
        background: icons.base(icons.loading.icon())
    }
};