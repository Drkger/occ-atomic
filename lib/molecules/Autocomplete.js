import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';

import Input from '../elements/Input';
import Droplist from '../elements/Droplist';

const styles = {
    droplist: {
        marginTop: '-31px'
    }
};

class Autocomplete extends React.Component {
    
    constructor(props) {
        super(props);
        this.state = {
            value: props.valueProp ? props.valueProp : '',
            focus: false
        };
        this.onChange = this.onChange.bind(this);
        this.onFocus = this.onFocus.bind(this);
        this.onBlur = this.onBlur.bind(this);
        this.onClick = this.onClick.bind(this);
    }
    
    onChange(value) {
        const { onChange } = this.props;
        this.setState({
            value, focus: true
        });
        if (onChange) onChange(value);
    }
    
    onFocus() {
        const { onFocus } = this.props;
        this.setState({
            focus: true
        });
        if (onFocus) onFocus();
    }
    
    onBlur() {
        const { onBlur } = this.props;
        setTimeout(() => {
            this.setState({
                focus: false
            });
        }, 200);
        if (onBlur) onBlur();
    }
    
    onClick(item) {
        const { onClick, onChange, itemTextKey } = this.props;
        this.setState({
            value: item[itemTextKey], focus: false
        });
        if (onClick) onClick(item);
        if (onChange) onChange(item[itemTextKey]);
    }
    
    render() {
        const { value, focus } = this.state;
        const { classes, valueProp, type, name, placeholder, items, groups, groupNameKey, groupIdKey, groupItemsKey, itemTextKey, itemTextRightKey, itemIdKey, onClick, returnItem } = this.props;
        return (
            <div>
                <Input
                    type={type}
                    name={name}
                    placeholder={placeholder}
                    onChange={this.onChange}
                    onFocus={this.onFocus}
                    onBlur={this.onBlur}
                    valueProp={value} />
                {(value && focus) && (
                    <Droplist
                        className={classes.droplist}
                        items={items}
                        groups={groups}
                        groupNameKey={groupNameKey}
                        groupIdKey={groupIdKey}
                        groupItemsKey={groupItemsKey}
                        itemTextKey={itemTextKey}
                        itemTextRightKey={itemTextRightKey}
                        itemIdKey={itemIdKey}
                        term={value}
                        onClick={this.onClick} />
                    )}
            </div>
        );
    }
}

Autocomplete.propTypes = {
    classes: PropTypes.object,
    valueProp: PropTypes.string,
    onChange: PropTypes.func,
    onFocus: PropTypes.func,
    onBlur: PropTypes.func,
    onClick: PropTypes.func,
    type: PropTypes.string,
    name: PropTypes.string,
    placeholder: PropTypes.string,
    items: PropTypes.array,
    groups: PropTypes.bool,
    groupNameKey: PropTypes.string,
    groupIdKey: PropTypes.string,
    groupItemsKey: PropTypes.string,
    itemTextKey: PropTypes.string,
    itemTextRightKey: PropTypes.string,
    itemIdKey: PropTypes.string
};

export default injectSheet(styles)(Autocomplete);