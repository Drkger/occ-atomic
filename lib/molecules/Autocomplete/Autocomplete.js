import React from 'react';
import PropTypes from 'prop-types';
import injectSheet from 'react-jss';

import Input from '../Input';
import Droplist from '../Droplist';
import Label from '../../elements/Label';

class Autocomplete extends React.Component {
    
    constructor(props) {
        super(props);
        this.state = {
            value: props.valueProp ? props.valueProp : '',
            focus: false,
            labels: []
        };
        this.onChange = this.onChange.bind(this);
        this.onFocus = this.onFocus.bind(this);
        this.onBlur = this.onBlur.bind(this);
        this.onClick = this.onClick.bind(this);
        this.addLabel = this.addLabel.bind(this);
        this.removeLabel = this.removeLabel.bind(this);
        this.removeSelected = this.removeSelected.bind(this);
        this.labels = [];
    }
    
    onChange(value) {
        const { onChange } = this.props;
        this.setState({
            value, focus: true
        });
        if (onChange) onChange(value);
    }
    
    onFocus() {
        const { onFocus } = this.props;
        this.setState({
            focus: true
        });
        if (onFocus) onFocus();
    }
    
    onBlur() {
        const { onBlur } = this.props;
        setTimeout(() => {
            this.setState({
                focus: false
            });
        }, 200);
        if (onBlur) onBlur();
    }
    
    onClick(item) {
        const { onClick, onChange, itemTextKey, stackLabels } = this.props;
        const child = this.child;
        if (stackLabels) {
            this.child.onClear();
            this.setState({
                value: ''
            });
            this.child.child.focus();
            this.addLabel(item);
        } else {
            this.setState({
                value: item[itemTextKey], focus: false
            });
            if (onChange) onChange(item[itemTextKey]);
        }
        if (onClick) onClick(stackLabels ? this.labels : item);
    }
    
    addLabel(item) {
        this.labels.push(item);
        this.setState({labels: this.labels});
    }
    removeLabel(item) {
        const { itemIdKey, onClick } = this.props;
        let pos = this.labels.map(label => { return label[itemIdKey];}).indexOf(item[itemIdKey]);
        this.labels.splice(pos, 1);
        this.setState({labels: this.labels});
        if (onClick) onClick(this.labels);
    }
    
    removeSelected() {
        const { labels } = this.state;
        const { items, itemIdKey, groups, groupItemsKey } = this.props;
        let cleanItems = [];
        if (groups) {
            items.map(group => {
                let cleanGroup = group[groupItemsKey].filter(item => {
                    const inLabels = labels.filter(label => {
                        return label[itemIdKey] == item[itemIdKey];
                    });
                    return !inLabels.length;
                });
                const modifiedGroup = Object.assign({}, group);
                modifiedGroup[groupItemsKey] = cleanGroup;
                if (cleanGroup.length) cleanItems.push(modifiedGroup);
            });
        } else {
            cleanItems = items.filter(item => {
                const inLabels = labels.filter(label => {
                    return label[itemIdKey] == item[itemIdKey];
                });
                return !inLabels.length;
            });
        }
        return cleanItems;
    }
    
    render() {
        const { value, focus, labels } = this.state;
        const { classes, id, type, name, label, placeholder, iconName, selectOnFocus, clear, items, groups, groupNameKey, groupIdKey, groupItemsKey, itemTextKey, itemTextRightKey, itemIdKey, onClick, stackLabels, limit, limitPlaceholder, style } = this.props;
        const cleanItems = this.removeSelected();
        return (
            <div className={classes.autoComplete} id={id} style={style}>
                <Input
                    onRef={ref => { this.child = ref; }}
                    type={type}
                    name={name}
                    label={label}
                    placeholder={((stackLabels && limit) && this.labels.length >= limit) ? limitPlaceholder : placeholder}
                    onChange={this.onChange}
                    onFocus={this.onFocus}
                    onBlur={this.onBlur}
                    selectOnFocus={selectOnFocus}
                    clear={clear}
                    disabled={(stackLabels && limit) && this.labels.length >= limit}
                    valueProp={value}
                    iconName={iconName} />
                {(value && focus) && (
                    <Droplist
                        className={classes.droplist}
                        items={cleanItems}
                        groups={groups}
                        groupNameKey={groupNameKey}
                        groupIdKey={groupIdKey}
                        groupItemsKey={groupItemsKey}
                        itemTextKey={itemTextKey}
                        itemTextRightKey={itemTextRightKey}
                        itemIdKey={itemIdKey}
                        term={value}
                        onClick={this.onClick} />
                )}
                {(stackLabels && labels.length) ? (
                    <div className={classes.labels}>
                        {
                            labels.map(label => (
                                <Label key={label[itemIdKey]} onClose={() => {this.removeLabel(label);}}>{label[itemTextKey]}</Label>
                            ))
                        }
                    </div>
                ) : ''}
            </div>
        );
    }
}

Autocomplete.propTypes = {
    classes: PropTypes.object,
    id: PropTypes.string,
    valueProp: PropTypes.string,
    onChange: PropTypes.func,
    onFocus: PropTypes.func,
    onBlur: PropTypes.func,
    onClick: PropTypes.func,
    type: PropTypes.string,
    name: PropTypes.string,
    label: PropTypes.string,
    placeholder: PropTypes.string,
    iconName: PropTypes.string,
    selectOnFocus: PropTypes.bool,
    clear: PropTypes.bool,
    items: PropTypes.array,
    groups: PropTypes.bool,
    groupNameKey: PropTypes.string,
    groupIdKey: PropTypes.string,
    groupItemsKey: PropTypes.string,
    itemTextKey: PropTypes.string,
    itemTextRightKey: PropTypes.string,
    itemIdKey: PropTypes.string,
    stackLabels: PropTypes.bool,
    limit: PropTypes.number,
    limitPlaceholder: PropTypes.string,
    style: PropTypes.object
};

export default Autocomplete;