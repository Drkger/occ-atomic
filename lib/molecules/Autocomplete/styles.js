export default {
    autoComplete: {
        position:'relative'
    },
    droplist: {
        position:'absolute',
        zIndex:'2',
        top:'49px',
        left:'0',
        width:'100%'
    },
    labels: {
        marginTop:'-18px',
        width:'100%'
    }
};