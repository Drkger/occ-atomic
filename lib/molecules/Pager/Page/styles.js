import colors from '../../../subatomic/colors';

export default {
    li: {
        display:'inline-block',
        cursor:'pointer',
        padding:'2px 11px',
        marginLeft:'4px',
        marginRight:'4px',
        borderRadius:'13px',
        cursor:'pointer',
        transition:'0.3s all',
        outline:'0',
        '&:hover': {
            background:'#ECECEE'
        }
    },
    active: {
        background:`${colors.blue} !important`,
        color:colors.white
    }
};