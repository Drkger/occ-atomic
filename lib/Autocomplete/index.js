import injectSheet from 'react-jss';
import Autocomplete from './Autocomplete';
import styles from './styles';

export default injectSheet(styles)(Autocomplete);