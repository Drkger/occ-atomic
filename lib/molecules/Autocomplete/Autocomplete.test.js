import React from 'react';
import { shallow } from 'enzyme';
import Autocomplete from './Autocomplete';
import styles from './styles';
import AutocompleteJSS from './index';

const reduceClasses = (prev, curr) => Object.assign({}, prev, { [curr]: curr });

const classes = Object.keys(styles).reduce(reduceClasses, {});

describe("Autocomplete", () => {

    it('matches the snapshot', () => {
        const items = [
            {
                text: 'Administrador',
                textRight: '(222)'
            },
            {
                text: 'Administrador Sr',
                textRight: '(134)'
            },
            {
                text: 'Administrador Jr',
                textRight: '(52)'
            },
            {
                text: 'Administrar',
                textRight: '(6)'
            },
        ];
        const wrapper = shallow(
            <Autocomplete items={items} classes={classes} />
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("AutocompleteJSS", () => {

    it('matches the snapshot', () => {
        const items = [
            {
                text: 'Administrador',
                textRight: '(222)'
            },
            {
                text: 'Administrador Sr',
                textRight: '(134)'
            },
            {
                text: 'Administrador Jr',
                textRight: '(52)'
            },
            {
                text: 'Administrar',
                textRight: '(6)'
            },
        ];
        const wrapper = shallow(
            <AutocompleteJSS items={items} />
        );
        expect(wrapper).toMatchSnapshot();
    });

});

describe("Autocomplete styles", () => {

    it('matches the snapshot', () => {
        expect(styles).toMatchSnapshot();
    });

});