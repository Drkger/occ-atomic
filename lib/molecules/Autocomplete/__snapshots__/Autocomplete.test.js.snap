// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Autocomplete matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Autocomplete
    classes={
      Object {
        "autoComplete": "autoComplete",
        "droplist": "droplist",
        "labels": "labels",
      }
    }
    items={
      Array [
        Object {
          "text": "Administrador",
          "textRight": "(222)",
        },
        Object {
          "text": "Administrador Sr",
          "textRight": "(134)",
        },
        Object {
          "text": "Administrador Jr",
          "textRight": "(52)",
        },
        Object {
          "text": "Administrar",
          "textRight": "(6)",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Jss(Input)
          autoFocus={false}
          clear={false}
          counter={false}
          disabled={false}
          iconName={undefined}
          label={undefined}
          maxLength={100}
          name={undefined}
          onBlur={[Function]}
          onChange={[Function]}
          onFocus={[Function]}
          onRef={[Function]}
          placeholder={undefined}
          selectOnFocus={false}
          type="text"
          valueProp=""
        />,
        "",
        "",
      ],
      "className": "autoComplete",
      "id": undefined,
      "style": undefined,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "autoFocus": false,
          "clear": false,
          "counter": false,
          "disabled": false,
          "iconName": undefined,
          "label": undefined,
          "maxLength": 100,
          "name": undefined,
          "onBlur": [Function],
          "onChange": [Function],
          "onFocus": [Function],
          "onRef": [Function],
          "placeholder": undefined,
          "selectOnFocus": false,
          "type": "text",
          "valueProp": "",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "",
      "",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Jss(Input)
            autoFocus={false}
            clear={false}
            counter={false}
            disabled={false}
            iconName={undefined}
            label={undefined}
            maxLength={100}
            name={undefined}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onRef={[Function]}
            placeholder={undefined}
            selectOnFocus={false}
            type="text"
            valueProp=""
          />,
          "",
          "",
        ],
        "className": "autoComplete",
        "id": undefined,
        "style": undefined,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "autoFocus": false,
            "clear": false,
            "counter": false,
            "disabled": false,
            "iconName": undefined,
            "label": undefined,
            "maxLength": 100,
            "name": undefined,
            "onBlur": [Function],
            "onChange": [Function],
            "onFocus": [Function],
            "onRef": [Function],
            "placeholder": undefined,
            "selectOnFocus": false,
            "type": "text",
            "valueProp": "",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "",
        "",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`AutocompleteJSS matches the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Jss(Autocomplete)
    items={
      Array [
        Object {
          "text": "Administrador",
          "textRight": "(222)",
        },
        Object {
          "text": "Administrador Sr",
          "textRight": "(134)",
        },
        Object {
          "text": "Administrador Jr",
          "textRight": "(52)",
        },
        Object {
          "text": "Administrar",
          "textRight": "(6)",
        },
      ]
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "classes": Object {
        "autoComplete": "Autocomplete-autoComplete-0-1-1",
        "droplist": "Autocomplete-droplist-0-1-2",
        "labels": "Autocomplete-labels-0-1-3",
      },
      "items": Array [
        Object {
          "text": "Administrador",
          "textRight": "(222)",
        },
        Object {
          "text": "Administrador Sr",
          "textRight": "(134)",
        },
        Object {
          "text": "Administrador Jr",
          "textRight": "(52)",
        },
        Object {
          "text": "Administrar",
          "textRight": "(6)",
        },
      ],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "classes": Object {
          "autoComplete": "Autocomplete-autoComplete-0-1-1",
          "droplist": "Autocomplete-droplist-0-1-2",
          "labels": "Autocomplete-labels-0-1-3",
        },
        "items": Array [
          Object {
            "text": "Administrador",
            "textRight": "(222)",
          },
          Object {
            "text": "Administrador Sr",
            "textRight": "(134)",
          },
          Object {
            "text": "Administrador Jr",
            "textRight": "(52)",
          },
          Object {
            "text": "Administrar",
            "textRight": "(6)",
          },
        ],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
